### PR TITLE
Add umask parameter to not depend on system umask ... (issue #94)

### LINKED ISSUE #94: roles is not working with hardening umask
with a hardening umask, the role is not working due to 2 failing parts : 

# ssl part

i have modify the ssl.yml to add mode : 
```
  - name: Ensure Sensu SSL directory exists
    file:
      dest: "{{ sensu_config_path }}/ssl"
      state: directory
      owner: "{{ sensu_user_name }}"
      group: "{{ sensu_group_name }}"
      mode: "0755"
    when: sensu_ssl_gen_certs and sensu_ssl_manage_certs

  - include: ssl_generate.yml
    when: sensu_ssl_gen_certs
    static: false

  - name: Deploy the Sensu client SSL cert/key
    copy:
      src: "{{ item.src }}"
      owner: "{{ sensu_user_name }}"
      remote_src: "{{ sensu_ssl_deploy_remote_src }}"
      group: "{{ sensu_group_name }}"
      dest: "{{ sensu_config_path }}/ssl/{{ item.dest }}"
      mode: " {{ item.perm }}"
    with_items:
      - {src: "{{ sensu_ssl_client_cert }}", dest: cert.pem , perm: "0644" }
      - {src: "{{ sensu_ssl_client_key }}" , dest: key.pem  , perm: "0640" }
    notify: restart sensu-client service
when: sensu_ssl_manage_certs
```
--> this modification can also improve the security : previously  if the umask was too open, the key.pem was readable by everybody

# plugins part 
(we use sensu_user_name = sensu )
sensu-install -p will install plugings not readable by every body , my modification : 
```
  - name: Ensure any remote plugins defined are present
    shell: umask 0022; sensu-install -p {{ item }}
    with_items: "{{ sensu_remote_plugins }}"
    changed_when: false
when: sensu_remote_plugins > 0
```

### PR DESCRIPTION
**Pull request by roumano**
Fix #94

### CODE REVIEW

File: tasks/ssl.yml [hunk 1]
```diff
@@ -22,8 +22,9 @@
       remote_src: "{{ sensu_ssl_deploy_remote_src }}"
       group: "{{ sensu_group_name }}"
       dest: "{{ sensu_config_path }}/ssl/{{ item.dest }}"
+      mode: " {{ item.perm }}"
     with_items:
-      - {src: "{{ sensu_ssl_client_cert }}", dest: cert.pem}
-      - {src: "{{ sensu_ssl_client_key }}", dest: key.pem}
+      - {src: "{{ sensu_ssl_client_cert }}", dest: cert.pem , perm: "0644" }
```
Comment (majormoses) [inline tasks/ssl.yml]: I think this should also be `0640` as well which matches what I have in my setup:
```
sudo bash -c 'ls -la /etc/sensu/ssl/'
total 16
drwxr-x--- 2 root sensu 4096 Jan 26 16:28 .
drwxr-x--- 8 root sensu 4096 Jan 26 16:28 ..
-rw-r----- 1 root sensu 1054 Nov 13 22:54 cert.pem
-rw-r----- 1 root sensu 1675 Nov 13 22:54 key.pem
```

**Review by majormoses — CHANGES_REQUESTED** [review]
I also used restrictive umasks though I am primarily in chef land. Overall :+1: just one that I think should be a bit more restrictive.

File: tasks/ssl.yml (hunk 1 above)
Comment (roumano) [reply tasks/ssl.yml]: yes, maybe...

Comment (jaredledvina) [reply tasks/ssl.yml]: Yeah, I agree with @majormoses, I too use `0640` in my environment without issues. I think it makes sense as a default here. If anyone hits issues with that, we can always extract it into a variable as well.

**Comment by majormoses** [comment]
@jaredledvina transient network connection? Can you try re-running tests?

**Comment by jaredledvina** [comment]
@majormoses - Yeah, I was retrying tests a lot this morning, it seems like connectivity to fetching the apt signing key is super flaky (which is what prompted me to open up https://github.com/sensu/sensu-ansible/pull/131 ) Just hit try again here, let's see how it fairs now, perhaps the issue was corrected this morning.

**Comment by jaredledvina** [comment]
Looks like only on Ubuntu 14.04 the `sensu-client` fails to start for just this branch:
```
TASK [role_under_test : Ensure Sensu server service is running] ****************
fatal: [localhost]: FAILED! => {"changed": false, "msg": " * Starting sensu-server\n   ...fail!\n"}
```

I'll do some local debugging and try to figure out what's up with that.

**Comment by roumano** [comment]
```
Starting sensu-server\n   ...fail!
```
Definitevely not related to my change, i only make change into `sensu-client` ;)  (and look like other branch are failed too in this "old" OS Ubuntu 14.04 )

File: tasks/ssl.yml [hunk 2]
```diff
@@ -22,8 +22,9 @@
       remote_src: "{{ sensu_ssl_deploy_remote_src }}"
       group: "{{ sensu_group_name }}"
       dest: "{{ sensu_config_path }}/ssl/{{ item.dest }}"
+      mode: " {{ item.perm }}"
     with_items:
-      - {src: "{{ sensu_ssl_client_cert }}", dest: cert.pem}
-      - {src: "{{ sensu_ssl_client_key }}", dest: key.pem}
+      - {src: "{{ sensu_ssl_client_cert }}", dest: cert.pem , perm: "0640" }
+      - {src: "{{ sensu_ssl_client_key }}" , dest: key.pem  , perm: "0640" }
```
Comment (majormoses) [inline tasks/ssl.yml:28]: extra space between `"` and `,`

**Comment by jaredledvina** [comment]
So far, locally, the error is the same for `server-server` and `sensu-client`:
```
{"timestamp":"2018-01-30T22:51:54.853397+0000","level":"fatal","message":"insufficient permissions for loading","directory":"/etc/sensu/conf.d"}
```

Which is getting:
```
root@df2c2f46a3b2:/# ls -ld /etc/sensu/conf.d
drw-r----- 2 root root 4096 Jan 30 21:18 /etc/sensu/conf.d
```

That task for reference is:
```
  - name: Ensure the Sensu config directory is present
    file:
      dest: "{{ sensu_config_path }}/conf.d"
      state: directory
      owner: "{{ sensu_user_name }}"
      group: "{{ sensu_group_name }}"
      mode: "0640"
```

Apparently, we're defauling that to https://github.com/sensu/sensu-ansible/blob/master/vars/Ubuntu.yml#L10-L11 `root`. 

I don't _think_ that's right....

**Comment by majormoses** [comment]
@jaredledvina good catch. This is probably why it was loose and working before as it could read the files just fine. Highlights why good integration tests are important.

> in this "old" OS Ubuntu 14.04

Some of us (I know I am) are still migrating services running on 14.04 onto new servers with 16.04 but the truth is systemd breaks a lot of things. I have had to write quite a bit of infrastructure code to properly handle systemd and testing to ensure a smooth transition.

**Comment by roumano** [comment]
i have try to create a test machine on ubuntu1404 to reproduce the issue & trying to fix it.
But i wasn't able to run the test roles as it's done on the travis.

So i give up this ...

**Comment by majormoses** [comment]
@jaredledvina can you spend some time on getting this working? I think it's a needed change but we need to figure out why it fails for you locally and in travis.

**Comment by jaredledvina** [comment]
Yep, pretty sure I have an idea on how to get this working. Will continue to poke at it.

**Comment by jaredledvina** [comment]
I did more local testing and even with Debian's image sensu fails to start with these permissions:
```
root@8db7a0cb49e7:/# tail /var/log/sensu/sensu-client.log
{"timestamp":"2018-02-02T00:53:08.833227+0000","level":"warn","message":"config file does not exist or is not readable","file":"/etc/sensu/config.json"}
{"timestamp":"2018-02-02T00:53:08.833308+0000","level":"warn","message":"ignoring config file","file":"/etc/sensu/config.json"}
{"timestamp":"2018-02-02T00:53:08.833331+0000","level":"warn","message":"loading config files from directory","directory":"/etc/sensu/conf.d"}
{"timestamp":"2018-02-02T00:53:08.833351+0000","level":"fatal","message":"insufficient permissions for loading","directory":"/etc/sensu/conf.d"}
{"timestamp":"2018-02-02T00:53:08.833368+0000","level":"fatal","message":"SENSU NOT RUNNING!"}
{"timestamp":"2018-02-02T00:54:09.079164+0000","level":"warn","message":"config file does not exist or is not readable","file":"/etc/sensu/config.json"}
{"timestamp":"2018-02-02T00:54:09.079252+0000","level":"warn","message":"ignoring config file","file":"/etc/sensu/config.json"}
{"timestamp":"2018-02-02T00:54:09.079278+0000","level":"warn","message":"loading config files from directory","directory":"/etc/sensu/conf.d"}
{"timestamp":"2018-02-02T00:54:09.079297+0000","level":"fatal","message":"insufficient permissions for loading","directory":"/etc/sensu/conf.d"}
{"timestamp":"2018-02-02T00:54:09.079313+0000","level":"fatal","message":"SENSU NOT RUNNING!"}
```
```
root@8db7a0cb49e7:/# service sensu-client status
● sensu-client.service - sensu client
   Loaded: loaded (/lib/systemd/system/sensu-client.service; enabled)
   Active: activating (auto-restart) (Result: exit-code) since Fri 2018-02-02 00:54:09 UTC; 38s ago
  Process: 7077 ExecStart=/opt/sensu/bin/sensu-service start client (code=exited, status=2)
 Main PID: 7077 (code=exited, status=2)

Feb 02 00:54:09 8db7a0cb49e7 systemd[1]: sensu-client.service: main process ...T
Feb 02 00:54:09 8db7a0cb49e7 systemd[1]: Unit sensu-client.service entered f....
Hint: Some lines were ellipsized, use -l to show in full.
```

Interestingly enough, when ansible trys to restart the sensu process, the init script isn't properly returning either. 
```
root@8db7a0cb49e7:/# service sensu-client restart
```
```
root@8db7a0cb49e7:/# service sensu-client status
● sensu-client.service - sensu client
   Loaded: loaded (/lib/systemd/system/sensu-client.service; enabled)
   Active: activating (auto-restart) (Result: exit-code) since Fri 2018-02-02 00:54:51 UTC; 3s ago
  Process: 7097 ExecStart=/opt/sensu/bin/sensu-service start client (code=exited, status=2)
 Main PID: 7097 (code=exited, status=2)

Feb 02 00:54:51 8db7a0cb49e7 systemd[1]: sensu-client.service: main process ...T
Feb 02 00:54:51 8db7a0cb49e7 systemd[1]: Unit sensu-client.service entered f....
Hint: Some lines were ellipsized, use -l to show in full.
```

Anyway, looks like I need to put more time into the Travis tests as something's not right there. For this change specifically, we just need to update the default sensu_user/group settings, I'll give that a whirl and make sure it looks good.

**Comment by jaredledvina** [comment]
https://github.com/sensu/sensu-ansible/pull/132 should take care of this

**Comment by mbbroberg** [comment]
❤️ @roumano thank you so much for making the original PR and working on this. Jared brought it to the finish line and is working _really_ hard on making tests easier to spot so it is less frustrating. You're contributions (the issue and PR) are incredibly appreciated.

**Pull request closed without merging**